cupy/linalg/solve.py

```diff
@@ -294,6 +297,80 @@ def inv(a):
     return b
 
 
+def _batched_inv(a):
+
+    assert(a.ndim >= 3)
+    a_shape = a.shape
+    a = a.copy().reshape(-1, a_shape[-2], a_shape[-1])
```

Comment:
```suggestion
    # to make `a` contiguous and to prevent `a` to be overwritten
    a = cupy.array(a, order='C').reshape(-1, a_shape[-2], a_shape[-1])
```
if `a` has to be C-contiguous later.

Reply:
I think either C-contiguous or F-contiguous is OK.

Reply:
Sorry. I wasn't aware of the default `ndarray.copy(order='C')`.

My comment looks valid because the test fails if
```python
    a = a.copy(order='K').reshape(-1, a_shape[-2], a_shape[-1])
```
with the input
```python
        a_gpu = cupy.asarray(a_cpu.T, order='C').T
```
, where `a_gpu` is NumPy's F-contiguous but each matrix `a_gpu[i]` is not.

Reply:
Ah, you are right. `a` must be C-contigous. It is a 3D (or more dimensional) array, not a 2D array. I forgot that...
Anyway, copy() is good enough.